hardhat.config.ts

```diff
@@ -95,6 +95,15 @@ const config: HardhatUserConfig = {
           browserURL: "https://sepolia.arbiscan.io/",
         },
       },
+      {
+        network: "polygonAmoy",
+        chainId: 80002,
+        urls: {
+          apiURL:
+            "https://www.oklink.com/api/explorer/v1/polygonamoy/contract/verify/async",
```

Comment:
i found this on some SO issue... might not be right

Reply:
ah, yeah should be `https://www.oklink.com/api/v5/explorer/contract/verify-source-code-plugin/AMOY_TESTNET`. still can't get it to work tho:
```
Verifying on 'polygon-amoy'...
Verifying implementation: 0x3BDC3E46d8de7A2e67A07Ac917382dB6D94577dc
Successfully submitted source code for contract
contracts/TablelandTables.sol:TablelandTables at 0x3BDC3E46d8de7A2e67A07Ac917382dB6D94577dc
for verification on the block explorer. Waiting for verification result...

We tried verifying your contract TablelandTables without including any unrelated one, but it failed.
Trying again with the full solc input used to compile and deploy it.
This means that unrelated contracts may be displayed on Etherscan...

Successfully submitted source code for contract
contracts/TablelandTables.sol:TablelandTables at 0x3BDC3E46d8de7A2e67A07Ac917382dB6D94577dc
for verification on the block explorer. Waiting for verification result...

Failed to verify implementation contract at 0x3BDC3E46d8de7A2e67A07Ac917382dB6D94577dc: The contract verification failed.
Reason: Fail - Unable to verify

Verifying proxy: 0x170fb206132b693e38adFc8727dCfa303546Cec1
Error: Etherscan API call failed with status 500, response: {"code":"50026","msg":"System error, please try again later.","data":[]}
```

Reply:
hmm, 500 error should mean it's an error on oklink's side, but who knows. is the `browserUrl` correct? 

did you by change compile some changes after deploying? https://github.com/NomicFoundation/hardhat/issues/1349#issuecomment-983002815

Reply:
hm, i didn't change the hardhat compiler version or any of the build settings, so i don't think it's that. but, if i change the compiler version to 0.8.20 and try to verify, it confirms that 0.8.19 _is_ the correct version for the contract i deployed:
> The contract you want to verify was compiled with solidity 0.8.19, but your configured compiler version is: 0.8.20.

based on the polygon discord, others are having the same issue with oklink. however, the polygon team noted [polygonscan will be supporting amoy](https://discord.com/channels/893151281848406016/1114418092601249943/1229463935275630735) within the next 2-3 weeks, so it's probably a moot point.

Reply:
cool. sg!

Reply:
https://amoy.polygonscan.com/ is up.

Reply:
@iPaulPro ahh nice, thanks. much better, will add